api/src/main/java/org/pmiops/workbench/ras/RasLinkService.java

```diff
@@ -144,6 +189,29 @@ private static String getLoginGovUsername(JsonNode userInfo) {
     return preferredUsername;
   }
 
+  /**
+   * Extracts user's eRA commons user id account from UserInfo response if has. See class javadoc
+   * for an example of eRA identity.
+   *
+   * <p>Returns empty if eRA is invalid or not linked.
+   */
+  public static Optional<String> getEraUserId(JsonNode userInfo) {
+    try {
+      String eRAUserId =
+          userInfo
+              .get(FEDERATED_IDENTITIES)
+              .get(IDENTITIES)
+              .get(0)
+              .get(ERA_COMMONS_PROVIDER_NAME)
+              .get(IDENTITY_USERID)
+              .asText();
+      return Optional.of(eRAUserId).filter(v -> !v.isEmpty());
+    } catch (NullPointerException e) {
```

Comment:
Catching NPE is generally not a great idea since it can hide programming errors. I suggest something like this instead ,and just checking for the optional presence at the end:

```
Optional.of(userInfo)
  .map(u -> u.get(FEDERATED_IDENTITIES))
  .map(u -> u..get(...))
```